lib/ansible/modules/remote_management/manageiq/manageiq_provider.py

```diff
@@ -0,0 +1,517 @@
+        description:
+          - How SSL certificates should be used for HTTPS requests. defaults to None.
+      certificate_authority:
+        required: false
+        default: null
+        description:
+          - The CA bundle string with custom certificates. defaults to None.
+'''
+
+EXAMPLES = '''
+  - name: Create a new provider in ManageIQ
+    manageiq_provider:
+      name: 'EngLab'
+      type: 'Openshift'
+      endpoints:
+        default:
+          auth_key: 'topSecret'
+          hostname: 'example.com'
+          port: 8443
+          verify_ssl: False
+        metrics:
+          hostname: 'example.com'
+          port: 443
+          verify_ssl: False
+      manageiq_connection:
+        url: 'http://127.0.0.1:3000'
+        username: 'admin'
+        password: 'smartvm'
+        verify_ssl: False
+
+  - name: Update an existing provider named 'EngLab'
+    manageiq_provider:
+      name: 'EngLab'
+      type: 'Openshift'
+      endpoints:
+        default:
+          auth_key: 'verySecret'
+          hostname: 'next.example.com'
+          port: 8443
+          verify_ssl: False
+        metrics:
+          hostname: 'next.example.com'
+          port: 443
+          verify_ssl: False
+      manageiq_connection:
+        url: 'http://127.0.0.1:3000'
+        username: 'admin'
+        password: 'smartvm'
+        verify_ssl: False
+
+  - name: Delete a provider in ManageIQ
+    manageiq_provider:
+      state: 'absent'
+      name: 'EngLab'
+      manageiq_connection:
+        url: 'http://127.0.0.1:3000'
+        username: 'admin'
+        password: 'smartvm'
+        verify_ssl: False
+'''
+
+RETURN = '''
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.manageiq import ManageIQ, manageiq_argument_spec
+
+
+def supported_providers():
+    return dict(
+        Openshift=dict(
+            class_name='ManageIQ::Providers::Openshift::ContainerManager',
+            authtype='bearer',
+            metrics_role='hawkular',
+            default_role='default',
+        ),
+        Amazon=dict(
+            class_name='ManageIQ::Providers::Amazon::CloudManager',
+            authtype='default',
+            metrics_role='metrics',
+            default_role='default',
+        ),
+    )
+
+
+def endpoint_list_spec():
+    return dict(
+        default=dict(required=True,
+                     type='dict', options=endpoint_argument_spec()),
+        metrics=dict(type='dict', options=endpoint_argument_spec()),
+    )
+
+
+def endpoint_argument_spec():
+    return dict(
+        role=dict(),
+        hostname=dict(),
+        port=dict(type='int'),
+        verify_ssl=dict(default=True, type='bool'),
+        certificate_authority=dict(),
+        security_protocol=dict(
+            choices=[
+              'ssl-with-validation',
+              'ssl-with-validation-custom-ca',
+              'ssl-without-validation',
+            ]),
+        ca_bundle_path=dict(),
```

Comment:
`ca_bundle_path` param was replaced by `certificate_authority`. Should be removed